updater/src/Updater/SyncLabels/Request.purs

```diff
@@ -48,6 +58,68 @@ type Sifted =
   , delete :: Array IssueLabel
   }
 
+listAllLabels :: String -> ExceptT Error Aff Unit
+listAllLabels token = do
+  let
+    allRepos = map { owner: "purescript", repo: _ } purescriptRepos
+      <> map { owner: "purescript-contrib", repo: _ } purescriptContribRepos
+      <> map { owner: "purescript-web", repo: _ } purescriptWebRepos
+      <>
+        map { owner: "purescript-node", repo: _ } purescriptNodeRepos
+
+    getRepoLabel = listLabel <<< (\{ owner, repo } -> { token, owner, repo })
+
+  results <- parTraverse getRepoLabel allRepos
+
+  let
+    handleInsert repo accMap label =
+      { labels: Map.insertWith (<>) label.name [ repo ] accMap.labels
+      , metadata: Map.insertWith Set.union label.name (Set.singleton { description: label.description, color: label.color }) accMap.metadata
+      }
+
+    f accMap { repo, labels } =
```

Comment:
`f` is a little hard for me to track here. Could we do `foldFn` instead?

Reply:
Specifically, we could do something like

```purs
  labelMap = foldl foldFn { labels: Map.empty, metadata: Map.empty } results
    where
    foldFn acc { repo, labels } = foldl (handleInsert repo) acc labels
    handleInsert repo m label =
      { labels:
          Map.insertWith append label.name [ repo ] m.labels
      , metadata: do
          let value = Set.singleton { description: label.description, color: label.color }
          Map.insertWith Set.union label.name m.metadata
      }
```

That way the only definition that ends up in general scope is the one we're using, `labelMap`.